test/OpenTelemetry.Instrumentation.Http.Tests/HttpClientTests.Basic.cs

```diff
@@ -300,30 +305,27 @@ public async Task HttpClientInstrumentation_AddViaFactory_HttpInstrumentation_Co
             Assert.IsType<Activity>(processor.Invocations[1].Arguments[0]);
         }
 
-        [Fact]
-        public async Task HttpClientInstrumentationBacksOffIfAlreadyInstrumented()
+        [Fact(Skip = "https://github.com/open-telemetry/opentelemetry-dotnet/issues/3729")]
+        public async Task HttpClientInstrumentationExportsSpansCreatedForRetries()
         {
-            // TODO: Investigate why this feature is required.
-            var processor = new Mock<BaseProcessor<Activity>>();
-
+            var exportedItems = new List<Activity>();
             var request = new HttpRequestMessage
             {
-                RequestUri = new Uri(this.url),
+                RequestUri = new Uri($"{this.url}retry"),
                 Method = new HttpMethod("GET"),
             };
 
-            request.Headers.Add("traceparent", "00-0123456789abcdef0123456789abcdef-0123456789abcdef-01");
-
-            using (Sdk.CreateTracerProviderBuilder()
+            using var traceprovider = Sdk.CreateTracerProviderBuilder()
                    .AddHttpClientInstrumentation()
-                   .AddProcessor(processor.Object)
-                   .Build())
-            {
-                using var c = new HttpClient();
-                await c.SendAsync(request);
-            }
+                   .AddInMemoryExporter(exportedItems)
+                   .Build();
 
-            Assert.Equal(4, processor.Invocations.Count); // SetParentProvider/OnShutdown/Dispose/OnStart called.
+            int maxRetries = 3;
+            using var c = new HttpClient(new RetryHandler(new HttpClientHandler(), maxRetries));
+            await c.SendAsync(request);
+
+            // number of exported spans should be 3(maxRetries)
+            Assert.Equal(3, exportedItems.Count());
```

Comment:
:100:

Reply:
also validate they have different span ids. (to rule out the case where have 3 dup span from the oiginal request)